Packs/cisco-meraki/Integrations/ciscomerakiv2/ciscomerakiv2.py

```diff
@@ -0,0 +1,515 @@
+    organization_id = args.get('organizationId')
+    res = client.fetch_organization_networks(organization_id=organization_id)
+    command_results = CommandResults(
+        outputs_prefix='Meraki.Networks',
+        outputs=res,
+        outputs_key_field='id',
+        readable_output=tableToMarkdown(f'Organization ({organization_id}) Networks:', res)
+    )
+    return_results(command_results)
+
+
+def meraki_fetch_network_devices_command(client: Client, args: dict):
+    network_id = args.get('networkId')
+    res = client.fetch_network_devices(network_id=network_id)
+    command_results = CommandResults(
+        outputs_prefix='Meraki.Devices',
+        outputs=res,
+        outputs_key_field='id',
+        readable_output=tableToMarkdown(f'Network ({network_id}) Devices:', res)
+    )
+    return_results(command_results)
+
+
+def meraki_fetch_device_command(client: Client, args: dict):
+    serial = args.get('serial')
+    res = client.fetch_device(serial=serial)
+    command_results = CommandResults(
+        outputs_prefix='Meraki.Devices',
+        outputs=res,
+        outputs_key_field='id',
+        readable_output=tableToMarkdown(f'Device (Serial: {serial}):', res)
+    )
+    return_results(command_results)
+
+
+def meraki_update_device_command(client: Client, args: dict):
+    serial = args.get('serial')
+    updates = {
+        'name': args.get('name'),
+        'tags': argToList(args.get('tags')),
+        'lat': args.get('lat'),
+        'lng': args.get('lng'),
+        'address': args.get('address'),
+        'notes': args.get('notes'),
+        'moveMapMarker': argToBoolean(args.get('moveMapMarker')),
+        'switchProfileId': args.get('switchProfileId'),
+        'floorPlanId': args.get('floorPlanId')
+    }
+    res = client.update_device(serial=serial, updates=updates)
+    command_results = CommandResults(
+        outputs_prefix='Meraki.Devices',
+        outputs=res,
+        outputs_key_field='serial',
+        readable_output=tableToMarkdown(f'Updated device (Serial: {serial}):', res)
+    )
+    return_results(command_results)
+
+
+def meraki_claim_device_command(client: Client, args: dict):
+    network_id = args.get('networkId')
+    serials = args.get('serials')
+    serial_list = argToList(serials)
+    res = client.claim_device(network_id=network_id, serials=serial_list)
+    outputs = {
+        "networkId": network_id,
+        "serials": serials,
+        'success': False,
+        'errors': None
+    }
+    readable_output = f'Claim of Serial numbers {serials} from network ID {network_id} failed'
+    if res.status_code == 204:
+        outputs['success'] = True
+        readable_output = f'Claim of Serial numbers {serials} from network ID {network_id} succeeded'
+    elif res.status_code == 404:
+        errors = extract_errors(res)
+        outputs['errors'] = errors
+        readable_output = f'Claim of Serial number {serials} from network ID {network_id} failed:\n{",".join(errors)}'
```

Comment:
What will be if one of the serial are invalid?
the entire operation will fail?
is it better to iterate over the list and claim each one 
and summarise the operation status?

Reply:
Usually I would agree, however the API documentation expects a list to be claimed. If I were to iterate, I would need to accept 40x codes as ok_codes and parse them separately into each device that failed. The error message returned should be clear enough that they can understand which devices would fail the claim command.